packages/vite/src/node/plugins/asset.ts

```diff
@@ -252,19 +253,51 @@ export async function urlToBuiltUrl(
   url: string,
   importer: string,
   config: ResolvedConfig,
-  pluginContext: PluginContext
+  ctx: PluginContext
 ): Promise<string> {
+  const builtUrl = await resolveBuiltUrl(url, config, ctx)
+  if (builtUrl) {
+    return builtUrl
+  }
+
   if (checkPublicFile(url, config)) {
     return config.base + url.slice(1)
   }
+
   const file = url.startsWith('/')
     ? path.join(config.root, url)
     : path.join(path.dirname(importer), url)
+
   return fileToBuiltUrl(
     file,
     config,
-    pluginContext,
+    ctx,
     // skip public check since we just did it above
     true
   )
 }
+
+async function resolveBuiltUrl(
+  url: string,
+  config: ResolvedConfig,
+  ctx: PluginContext
+) {
```

Comment:
```suggestion
): Promise<string | undefined> {
```

This will help to not come in conflict when we turn on `@typescript-eslint/explicit-module-boundary-types`